src/raft_server.cxx

```diff
@@ -1110,12 +1114,16 @@ bool raft_server::check_leadership_validity() {
     // Check if quorum is not responding.
     int32 num_voting_members = get_num_voting_members();
 
+
     int leadership_expiry = get_leadership_expiry();
-    int32 nr_peers = (int32)get_not_responding_peers_count();
-    if (leadership_expiry < 0) {
-        // Negative expiry: leadership will never expire.
-        nr_peers = 0;
+
+    int32 nr_peers{0};
+
+    // Negative expiry: leadership will never expire.
+    if(leadership_expiry > 0){
```

Comment:
Please put spaces too.
And the condition should be `if (leadership_expiry >= 0)`. `nr_peers` sets to `0` only when `leadership_expiry` is a negative number. If `leadership_expiry == 0`, it uses the default expiry (`response_limit_`).

Reply:
Thanks for the comment. Probably that is not the most clear part, but `get_leadership_expiry`() will return number greater than 0 if `params->leadership_expiry_` is set to 0.

Reply:
Yeah, but this if condition needs to be correct on its own, without relying on the result of other function. Otherwise it will be mistake-prone and potentially introduce a bug silently.